test/unit/services/sshAdapter_spec.js

```diff
@@ -0,0 +1,357 @@
+			var instanceStatusStub;
+			var result;
+			var sshAdapter = new SshAdapter(ec2);
+			before(function () {
+				instanceStatusStub = sinon.stub(ec2,"describeInstanceStatusAsync", function () {
+					var data = {
+						InstanceStatuses : [ {
+							SystemStatus : {
+								Details : [ {
+									Status : "insufficient-data"
+								} ]
+							}
+						} ]
+					};
+					return (Bluebird.resolve(data));
+				});
+				sshAdapter.canSsh(id)
+				.then(function (data) {
+					result = data;
+				})
+				.catch(function (err) {
+					result = err;
+				});
+			});
+
+			after(function () {
+				instanceStatusStub.restore();
+			});
+
+			it("it returns an error as there is insufficient-data", function () {
+				expect(instanceStatusStub.called).to.be.true;
+				expect(instanceStatusStub.args[ 0 ][ 0 ].InstanceIds[ 0 ]).to.equal(id);
+				expect(result).to.be.an.instanceof(Error);
+				expect(result.message).to.equal("System Checks Failed");
+			});
+		});
+
+	});
+
+	describe("Starts ssh Polling", function () {
+
+		describe("when system checks have passed", function () {
+			var instanceStatusStub;
+			var clock;
+			var result;
+			var sshAdapter = new SshAdapter(ec2);
+			before(function () {
+				clock = sinon.useFakeTimers();
+				instanceStatusStub = sinon.stub(ec2,"describeInstanceStatusAsync", function () {
+					var data = {
+						InstanceStatuses : [ {
+							SystemStatus : {
+								Details : [ {
+									Status : "passed"
+								} ]
+							}
+						} ]
+					};
+					return (Bluebird.resolve(data));
+				});
+				clock.tick(SSH_POLLING_INTERVAL * 1000);
+
+				sshAdapter.startSshPolling(id)
+				.then(function (data) {
+					result = data;
+				})
+				.catch(function (err) {
+					result = err;
+				});
+				clock.tick(SSH_POLLING_INTERVAL * 1000);
+			});
+
+			after(function () {
+				instanceStatusStub.restore();
+				clock.restore();
+			});
+
+			it("is able to connect", function () {
+				expect(instanceStatusStub.called).to.be.true;
+				expect(instanceStatusStub.args[ 0 ][ 0 ].InstanceIds[ 0 ]).to.equal(id);
+				expect(result).to.not.be.an.instanceof(Error);
+			});
+		});
+
+		describe("when there is no information regarding the status", function () {
+			var instanceStatusStub;
+			var clock;
+			var sshAdapter = new SshAdapter(ec2);
+			before(function () {
+				clock = sinon.useFakeTimers();
+				instanceStatusStub = sinon.stub(ec2, "describeInstanceStatusAsync", function () {
+					var data = {
+						InstanceStatuses : []
+					};
+					return (Bluebird.resolve(data));
+				});
+
+				clock.tick(SSH_POLLING_INTERVAL * 1000);
+				sshAdapter.startSshPolling(id);
+				clock.tick(SSH_POLLING_INTERVAL * 1000);
+			});
+
+			after(function () {
+				instanceStatusStub.restore();
+				clock.restore();
+			});
+
+			it("is not able to connect", function () {
+				expect(instanceStatusStub.called).to.be.true;
+				expect(instanceStatusStub.args[ 0 ][ 0 ].InstanceIds[ 0 ]).to.equal(id);
```

Comment:
How do we know that it is not able to connect? It seems like we only check the arguments here and not any kind of resulting state.